### DISCUSSION TITLE
API Tokens for non-customers

### DISCUSSION THREAD

**Discussion by cchurch132** [amberelectric/public-api#149] [General]
It looks like this isn't an option, but holding out hope it might be possible. I'm trying to simulate over a period of time, the difference between existing retailer and amber. I run a battery simulator in Home Assistant, fairly straight forward, but I'd like to use amber pricing as triggers to force discharging/ charging.

**Comment by madpilot** [comment]
Sorry - we don't offer an API for non-customers yet. You _can_ sign up, then cancel - we don't remove API access for closed accounts, and there is a 10-day cooling off period.

**Reply by cchurch132** [reply]
Thanks for the reply. Makes sense. Is the API access created instantly or does it take sometime to provision?

**Reply by Roving-Ronin** [reply]
@madpilot Hi Myles, just wondering if this is still the case re gaining access to Amber API / HA integration.  Non battery owner, so I'd probably not go good on Amber myself, but looking to add support for Amber in HA for a package allowing other users to make their inverter+batteries more useful.

**Reply by madpilot** [reply]
@Roving-Ronin What's the package?

**Reply by Roving-Ronin** [reply]
@madpilot Myles,
For the Huawei (now iStore in Australia) I've done the effort to support them in the SunSynk card ( https://github.com/Roving-Ronin/sunsynk-power-flow-card ), but now also looking to see if I can get an Australian version of Hauwei PEES ( https://github.com/JensenNick/huawei_solar_pees ) going instead of continuing my own version defining sensors to track where power/energy is going and tracking costs/savings etc. Idea being that users can then just install (be good if I can work out a HACS type package install, or else just copy the yaml files to /packages at worst) it and have:
- Graphical card to show power/energy flows live
- Entity card to show whats being used where / how much power or energy / costs and savings
- Ability to automate things like dump/export to grid (if they have a battery) if there's a good FIT rate or import and charge if its cheap to do so (something that Huawei / iStore doesn't have the ability to do normally, but can be done via the WLCRS integration that is leveraged for accessing the inverters/battery/meters etc.

**Reply by madpilot** [reply]
And you just want to test how the amberelectric HASS integration presents it's data?

**Reply by Roving-Ronin** [reply]
Obtain data from it, to allow testing and utilisation by custom sensors (i.e. get the above working with Amber HA) so Amber users can then use it / iStore users sign up to Amber for this.

**Reply by Roving-Ronin** [reply]
@madpilot any idea as to if dev accounts are catered for, or we still need to sign up and cancel (and account remains active) ? Thanks

**Reply by madpilot** [reply]
We don't at the moment. Email me at myles.eftos@amber.com.au and I can generate a token on a test account we have

**Reply by madpilot** [reply]
On leave this week, but I'll send it through next week when I'm back

**Comment by rob0101** [comment]
“I'd like to use amber pricing as triggers to force discharging/ charging.”

If you proceed with the model, make sure you account for the final 30m price.  For most of each 30m you’ll be acting based on estimated prices.
